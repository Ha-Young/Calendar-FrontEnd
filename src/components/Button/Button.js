import React from "react";

const Button = function ({ className, children, onClick }) {
  return (
    <button
      type="button"
      className={className}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
