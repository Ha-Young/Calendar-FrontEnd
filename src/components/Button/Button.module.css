.Button {
  width: 200px;
  height: 40px;
  margin-bottom: 50px;
  border: 1px solid #E5E8E8;
  border-radius: 20px;
  color: rgba(0, 0, 0, 0.6);
  font-size: 16px;
  text-align-last: center;
  text-transform: uppercase;
  transition: all 0.3s ease-in-out;
}

.Button:focus {
  border: 0;
  outline: 0;
}

.Button:hover {
  border: none;
  background-color: black;
  color: white;
  cursor: pointer;
}
