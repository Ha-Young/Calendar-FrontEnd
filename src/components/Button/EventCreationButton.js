import React from "react";
import { Link } from "react-router-dom";


function EventCreationButton({ onClick }) {
  return (
    <Link to="/events/new">
      <button onClick={onClick}>이벤트 생성</button>
    </Link>
  );
}

export default EventCreationButton;
