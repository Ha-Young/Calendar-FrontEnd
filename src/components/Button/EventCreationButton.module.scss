.Button {
  width: 6em;
  height: 3em;
  margin-left: 0;
  cursor: pointer;
  border-radius: 100px;
  outline: none;
}

