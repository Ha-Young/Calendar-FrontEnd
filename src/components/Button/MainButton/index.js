import MainButton from "components/Button/MainButton/MainButton";

export default MainButton;
