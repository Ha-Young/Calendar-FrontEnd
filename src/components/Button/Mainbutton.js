import React from "react";
import styles from "./Button.module.css";
import * as Aioutline from "react-icons/ai";

export default function mainButton({ onClick }) {
  return (
    <div className={`${styles.wrapper}`}>
      <Aioutline.AiFillPlusCircle onClick={onClick} />
    </div>
  );
}
