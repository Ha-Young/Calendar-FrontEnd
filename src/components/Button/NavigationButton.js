import React from "react";
import { FaChevronCircleLeft, FaChevronCircleRight } from "react-icons/fa";
import styles from "./NavigationButton.module.scss";

function NavigationButton({ onClick }) {
  return (
    <div className={styles.ButtonWrap}>
      <FaChevronCircleLeft
        onClick={(e) => onClick(e.target.id)}
        id="left"
        className={styles.Button}
      />
      <FaChevronCircleRight
        onClick={(e) => onClick(e.target.id)}
        id="right"
        className={styles.Button}
      />
    </div>
  );
}

export default NavigationButton;
