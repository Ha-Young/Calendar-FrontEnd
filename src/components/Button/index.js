import React from "react";
import styles from "./Button.module.css";

export default function Button({
  title,
  onClick,
}) {
  return (
    <button onClick={onClick} className={styles.Button} title={title}>
      {title}
    </button>
  );
}
