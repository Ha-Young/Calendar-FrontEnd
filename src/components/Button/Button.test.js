import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Button from "./Button";

afterEach(cleanup);

it("matches snapshot", () => {
  const utils = render(<Button title="Create Event" />);
  expect(utils.container).toMatchSnapshot();
});

it("shows the props correctly", () => {
  const utils = render(<Button title="Create Event" />);
  utils.getByText("Create Event");
});

it("shows the props correctly", () => {
  const utils = render(<Button title="Save" />);
  utils.getByText("Save");
});

it("works click event correctly", () => {
  const onClick = () => console.log("Click");
  const utils = render(<Button title="text" onClick={onClick} />);
  fireEvent.click(utils.getByText("text"), onClick);
});
