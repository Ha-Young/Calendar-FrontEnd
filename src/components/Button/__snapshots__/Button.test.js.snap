// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    class="Button"
  >
    Create Event
  </button>
</div>
`;
